Here is a story. March 24th, 2023 was the day my younger brother Max found out he did not get into his dream high school in San Francisco, Lowell High School. Max was disappointed, naturally. I felt bad for him because he studied really hard just to get rejected by his dream school. So I thought to myself, there's nothing I can do now to change the fact that he did not get into his first choice. But what if, I can make a difference in what colleges he could get into, because last thing I want to see Max doing is beating himself up so bad he gives up on school entirely. 

With that as motivation, I set out to collect information and data about college admissions. And I started the search at the University of California Systems -- the prestigious colleges of California at least half the price of a private institution.

And Here, are my findings.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

In [2]:
non = pd.read_excel("FR GPA by Year_non_CA.xlsx")
public = pd.read_excel("FR GPA by Year_public_CA.xlsx")
private = pd.read_excel("FR GPA by Year_private_CA.xlsx")

In [3]:
non['Type'] = "non CA"
public['Type'] = "public CA"
private['Type'] = "private CA"

In [4]:
df = pd.concat([public, private, non])

In [5]:
df = df.reset_index(drop= True)

In [6]:
df

Calculation1                        School  \
0           A B MILLER HIGH SCHOOL050944        A B MILLER HIGH SCHOOL   
1      ABRAHAM LINCOLN HIGH SCHOOL051520   ABRAHAM LINCOLN HIGH SCHOOL   
2      ABRAHAM LINCOLN HIGH SCHOOL052910   ABRAHAM LINCOLN HIGH SCHOOL   
3      ABRAHAM LINCOLN HIGH SCHOOL053075   ABRAHAM LINCOLN HIGH SCHOOL   
4     ACAD FOR ACADEMIC EXCELLENCE050126  ACAD FOR ACADEMIC EXCELLENCE   
...                                  ...                           ...   
4010          YORKTOWN HIGH SCHOOL336085          YORKTOWN HIGH SCHOOL   
4011          YORKTOWN HIGH SCHOOL470130          YORKTOWN HIGH SCHOOL   
4012          YOUNGKER HIGH SCHOOL030026          YOUNGKER HIGH SCHOOL   
4013     YUMA CATHOLIC HIGH SCHOOL030573     YUMA CATHOLIC HIGH SCHOOL   
4014       ZIONSVILLE COMMUNITY HS153900       ZIONSVILLE COMMUNITY HS   

                  City County/State/Country   App GPA   Adm GPA  Enrl GPA  \
0              Fontana       San Bernardino  3.659867  3.831091  3.971000   
1          Los Angeles          Los Angeles  3.634337  3.906604  3.882174   
2        San Francisco        San Francisco  3.964286  4.077178  4.143172   
3             San Jose          Santa Clara  4.063804  4.166441  4.242500   
4         Apple Valley       San Bernardino  3.803846  3.906500  3.937778   
...                ...                  ...       ...       ...       ...   
4010  Yorktown Heights             New York  3.924000       NaN       NaN   
4011         Arlington             Virginia  4.185172  4.270000       NaN   
4012           Buckeye              Arizona  3.470000       NaN       NaN   
4013              Yuma              Arizona  3.926923  4.227143       NaN   
4014        Zionsville              Indiana  4.104167       NaN       NaN   

           Type  
0     public CA  
1     public CA  
2     public CA  
3     public CA  
4     public CA  
...         ...  
4010     non CA  
4011     non CA  
4012     non CA  
4013     non CA  
4014     non CA  

[4015 rows x 8 columns]

In [7]:
df_city = df.groupby(['City'])[['App GPA', 'Adm GPA', 'Enrl GPA']].agg(['min', 'max', 'mean', 'var'])
df_city

App GPA                                 Adm GPA            \
                   min       max      mean       var       min       max   
City                                                                       
Aberdeen      4.240000  4.240000  4.240000       NaN       NaN       NaN   
Abington      4.005000  4.005000  4.005000       NaN       NaN       NaN   
Acton         3.827778  4.050000  3.916546  0.013843  3.908333  3.992000   
Ada           4.138571  4.138571  4.138571       NaN       NaN       NaN   
Addison       4.007647  4.007647  4.007647       NaN  4.207778  4.207778   
...                ...       ...       ...       ...       ...       ...   
Yucaipa       3.839588  3.839588  3.839588       NaN  4.007903  4.007903   
Yucca Valley  3.805600  3.805600  3.805600       NaN  3.953000  3.953000   
Yuma          3.740000  4.166364  3.991822  0.039435  4.181250  4.227143   
Zionsville    4.104167  4.104167  4.104167       NaN       NaN       NaN   
n/a           3.622000  4.226667  3.924333  0.182811       NaN       NaN   

                                  Enrl GPA                          
                  mean       var       min       max      mean var  
City                                                                
Aberdeen           NaN       NaN       NaN       NaN       NaN NaN  
Abington           NaN       NaN       NaN       NaN       NaN NaN  
Acton         3.960111  0.002047       NaN       NaN       NaN NaN  
Ada                NaN       NaN       NaN       NaN       NaN NaN  
Addison       4.207778       NaN       NaN       NaN       NaN NaN  
...                ...       ...       ...       ...       ...  ..  
Yucaipa       4.007903       NaN  4.106667  4.106667  4.106667 NaN  
Yucca Valley  3.953000       NaN  3.922857  3.922857  3.922857 NaN  
Yuma          4.204196  0.001053       NaN       NaN       NaN NaN  
Zionsville         NaN       NaN       NaN       NaN       NaN NaN  
n/a                NaN       NaN       NaN       NaN       NaN NaN  

[1681 rows x 12 columns]

As a San Franciscan, I want to know more about the big picture of high school students' admissions into UC's in the area. It's the first step to understanding Max's competition, at a local level.

In [8]:
sf = df[df['City'] == "San Francisco"]

I then also separated the San Francisco portion into public schools and private schools to see the difference between the two categories. I do this for a quite silly reason. But I had to find out.

The reason being is that as a general concensus among the Asian American community, especially amongst the parents, private schools are seen as more prestigious than and superior over public schools in many factors. Here are a few examples: One, resources. Two, the people and culture. And lastly, GRADES.

Resources: the parents claim that private schools have more recources for the students to succeed. For example more technologies and field trips. These factors MAY contribute to the success of a student, but are not essential to the likelihood of success.

People and culture: while the parents might have a point, surrounding yourself with better people will generally get you better in life, it is hard to quantify in the context of college admission. So we will disregard this for now because it plays a less significant role in the data.

Grades: As Asians, we pride ourselves in having good grades. We are even teased for having such good grades it's become a stereotype. Regardless of the validity of the stereotype, many Asian parents believe private school students have far better grades and in turn are destined to set their kids up for success. 
Now THIS, we can actually quantify, because we have data for it. Let's find out.

In [9]:
sf_public = sf[sf['Type'] == 'public CA']
sf_private = sf[sf['Type'] == 'private CA']

In [37]:
sfpublic_bar_graph = px.bar(sf_public, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
sfpublic_bar_graph.show()

In [38]:
sfprivate_bar_graph = px.bar(sf_private, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
sfprivate_bar_graph.show()

In [12]:
sf_public[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA   Enrl GPA
count  19.000000  18.000000  14.000000
mean    3.838389   3.940430   4.067277
std     0.164230   0.173136   0.137863
min     3.593333   3.585556   3.798000
25%     3.705351   3.847500   4.023014
50%     3.860000   3.977347   4.092681
75%     3.963393   4.059029   4.143520
max     4.159834   4.247803   4.279897

In [13]:
sf_private[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA   Enrl GPA
count  18.000000  16.000000  14.000000
mean    3.955987   4.028775   4.113885
std     0.139563   0.117402   0.067692
min     3.670000   3.753600   4.020000
25%     3.892426   3.971190   4.061026
50%     3.983368   4.053125   4.104176
75%     4.044778   4.112771   4.172750
max     4.128316   4.167273   4.226667

A little background information: Max got into George Washington High School in San Francisco. It is a San Francisco public school, same as Lowell High School.

The reason many kids in the San Francisco Unified School District wanted to get into Lowell High is because of the prestige. As a Lowell Alumni, I can attest to the situation. Lowell students are generally lightyears ahead of other San Francisco public schools, be it grades, sports, arts, robotics and other extracurriculars. We were always number 1 in any categorty compared to other schools. Hence the reputation. However, Max did not get into Lowell High School.

But, there is no other way to write a wrong than to look forward and do better the next time you have a chance. So here we are. trying to debunk some myths and find the optimal way for Max to get into a college he rightfully deserves, through hard work and determination.

As we can see above, the data shows miniscule difference in the application GPA, referring to the average GPA of the stack of applications UC's get. 

With that in mind, private schools cost money. A LOT of money. While public schools cost absolutely nothing, aside from occasional donations to the alumni associations and helping out at field trips. This begs the question: do the reputations of private schools outweigh the fact that everyone should pay for high school education because it guarantees success?

While it may be true, that students of private schools will receive better treatment on a day to day, how likely will the kids remember what they did in high school after they've gotten a job after college graduation? Very unlikely. 

The data clearly shows little to no difference in the admission GPA. Public school students score just as high, if not higher than private school students to get into a UC school. This further debunks the Asian parent myths and proves that it is possible to thrive in a public school AND go to a great college, such as a UC.

This brings me to the conclusion: Max needs to score at the minimum, around a 4.02 GPA to be able to get into a good college in the UC systems.

As we all know, all colleges are created equal, but unfortunately some, are more equal than the others.

UC's are no exception. UC Berkeley, the pinnacle of UC's, is a phenomenal college to have an undergraduate degree at. With this reputation, they get to pick at a much higher GPA requirement. So I went ahead and explored.

P.S. Max you're in a world of trouble, but you got this! It is doable, you just have to try really hard. Don't give up!

In [15]:
berk_public = pd.read_excel("FR GPA by Year berkeley public.xlsx")
berk_private = pd.read_excel("FR GPA by Year berkeley private.xlsx")

I want to focus on SF areas first to see how the competition stands.

In [17]:
berk_public_sf = berk_public[berk_public['City'] == "San Francisco"]
berk_private_sf = berk_private[berk_private['City'] == "San Francisco"]

In [39]:
fig1 = px.bar(berk_public_sf, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig1.show()

In [40]:
fig2 = px.bar(berk_private_sf, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig2.show()

In [19]:
berk_public_sf[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA   Adm GPA  Enrl GPA
count  18.000000  9.000000  8.000000
mean    3.891033  4.258440  4.230413
std     0.167808  0.050261  0.072123
min     3.565556  4.154194  4.097000
25%     3.788646  4.234800  4.214280
50%     3.920973  4.271176  4.240659
75%     4.021117  4.280000  4.254375
max     4.183755  4.338101  4.349000

In [20]:
berk_private_sf[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA  Enrl GPA
count  16.000000  10.000000  5.000000
mean    4.016390   4.217501  4.210482
std     0.149540   0.040456  0.039327
min     3.625106   4.162222  4.165385
25%     4.006902   4.193643  4.176000
50%     4.081984   4.209875  4.215882
75%     4.093478   4.246194  4.238000
max     4.148986   4.277500  4.257143

(Yeah people applying to Berkeley are pretty smart, you've got to try very hard, Max. There is no way around it.)

As seen above, the admission GPA shows how high the average GPA is according to the students that are admitted. It is a good reference point as to how well one has to perform in their respective high schools to have a chance to get into UC Berkeley.

I then wanted to explore down South, in the Santa Clara County, to see what other competitions look like.

In [32]:
berk_public_sc = berk_public[berk_public['County/State/Country'] == "Santa Clara"]
berk_private_sc = berk_private[berk_private['County/State/Country'] == "Santa Clara"]

In [45]:
fig4 = px.bar(berk_public_sc, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig4.show()

In [43]:
fig3 = px.bar(berk_private_sc, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig3.show()

In [35]:
berk_public_sc[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA   Enrl GPA
count  53.000000  36.000000  27.000000
mean    4.033408   4.289799   4.270511
std     0.109940   0.048307   0.049248
min     3.716316   4.186000   4.181333
25%     3.977222   4.250701   4.233421
50%     4.051282   4.292456   4.264091
75%     4.118235   4.331856   4.306094
max     4.228187   4.361750   4.366667

In [36]:
berk_private_sc[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA  Enrl GPA
count  18.000000  11.000000  7.000000
mean    4.100428   4.268857  4.253956
std     0.182009   0.031270  0.032828
min     3.832000   4.218000  4.216667
25%     4.026458   4.245500  4.233750
50%     4.106779   4.271429  4.242857
75%     4.135675   4.296607  4.268333
max     4.669091   4.302593  4.314000

In [47]:
berk_public_sm = berk_public[berk_public['County/State/Country'] == "San Mateo"]
berk_private_sm = berk_private[berk_private['County/State/Country'] == "San Mateo"]

In [48]:
fig10 = px.bar(berk_public_sm, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig10.show()

In [49]:
fig11 = px.bar(berk_private_sm, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig11.show()

In [50]:
berk_public_sm[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA   Enrl GPA
count  23.000000  16.000000  14.000000
mean    3.984474   4.276284   4.249358
std     0.131839   0.047910   0.052533
min     3.660000   4.136667   4.128000
25%     3.962494   4.252484   4.242083
50%     4.026667   4.297222   4.260060
75%     4.063212   4.308271   4.290000
max     4.147047   4.318182   4.296667

In [51]:
berk_private_sm[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA   Adm GPA  Enrl GPA
count  11.000000  7.000000  5.000000
mean    4.068810  4.224420  4.208984
std     0.091910  0.046731  0.046491
min     3.921000  4.136667  4.148889
25%     4.017798  4.208646  4.185556
50%     4.048462  4.236364  4.212857
75%     4.119075  4.252531  4.223333
max     4.244026  4.275556  4.274286

I also explored East Bay.

In [53]:
berk_public_a = berk_public[berk_public['County/State/Country'] == "Alameda"]
berk_private_a= berk_private[berk_private['County/State/Country'] == "Alameda"]

In [54]:
fig12 = px.bar(berk_public_a, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig12.show()

In [55]:
fig13 = px.bar(berk_private_a, x = "School", y = ["App GPA", "Adm GPA", "Enrl GPA"], barmode = 'group')
fig13.show()

In [56]:
berk_public_a[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA    Adm GPA   Enrl GPA
count  57.000000  31.000000  25.000000
mean    3.877477   4.249042   4.243906
std     0.227181   0.105379   0.096022
min     3.024118   3.928000   3.928000
25%     3.708235   4.205764   4.226000
50%     3.968060   4.279333   4.266154
75%     4.020323   4.317273   4.298571
max     4.165047   4.416000   4.363158

In [57]:
berk_private_a[['School', 'App GPA', 'Adm GPA', 'Enrl GPA']].describe()

App GPA   Adm GPA  Enrl GPA
count  16.000000  8.000000  4.000000
mean    3.978694  4.202051  4.176750
std     0.186760  0.059114  0.045778
min     3.654000  4.094000  4.120000
25%     3.857422  4.179643  4.161250
50%     4.003641  4.216364  4.177500
75%     4.127847  4.242438  4.193000
max     4.218000  4.265000  4.232000

In conclusion, Max has to score a fairly high GPA just to have a chance at an attempt to get into UC Berkeley. I am not saying he will fail, but he has got to try extra extra hard to have a chance. 

Max, I believe in you.

Jack